### PR TITLE
Adds support for Feature Middleware

### PR DESCRIPTION
**Pull request by JustSteveKing**
This PR adds the ability to enable Feature blocking middleware, where if a feature is not active or defined, a `HttpException` will be thrown. The middleware pipeline will continue if the feature is active or has been defined.

Example usage:

```php
Route::prefix('beta')->middleware(['feature:beta')->group();
```

Requests to anything under `beta/*` will run through this middleware and see if the feature has been defined.

You can also pass through a variadic list of features you wish to check against - if any of these are not active or are undefined, then an exception will be thrown.

```php
Route::prefix('beta')->middleware(['feature:beta,another')->group();
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,4 @@
 composer.lock
 /phpunit.xml
 .phpunit.result.cache
+.idea
```
Comment (driesvints) [inline .gitignore]: Can you add this to your global gitignore?

Comment (JustSteveKing) [reply .gitignore]: Resolved 🎉

**Comment by taylorotwell** [comment]
Renamed to `EnsureFeaturesAreActive`.

Added ability to configure failure responses:

```php
EnsureFeaturesAreActive::whenInactive(fn ($request, $features) => 'some-response');
```

**Comment by taylorotwell** [comment]
Thanks 👍

**Comment by taylorotwell** [comment]
Documented ✅

**Comment by JustSteveKing** [comment]
Yay! This is officially my first PR to one to something in the Ecosystem. Mission accomplished!

**Pull request merged**